### DISCUSSION TITLE
Process for choosing multiple dependency versions for formulae?

### DISCUSSION THREAD

**Discussion by gjtorikian** [Homebrew/discussions#4483] [Tap maintenance and Homebrew development] [answered]
### Output of `brew config`

```shell
n/a
```

### Output of `brew doctor`

```shell
n/a
```

### Description of issue

Apologies for taking advantage of our friendship and pinging you directly @MikeMcQuaid  but...you _were_ the last person to touch the code I'm asking about. 😄 

`postgresql@14` is currently marked as a hardcoded dependent of `postgis`:

https://github.com/Homebrew/homebrew-core/blob/8366c3fce428fe67764bdaa04345489b34d0ef42/Formula/postgis.rb#L38

Except, the latest version of Postgres is 15, and that means this formula won't run on Postgres 15.

Is there a pattern for accepting a list of `depends_on` for various versions of the same formula? I searched the docs and couldn't find any other result for this issue. I'd be more than happy to make a PR to update this line, if I learn how to update it!

**Comment by SMillerDev** [comment]
There is no way to make it accept multiple versions there

**Comment by MikeMcQuaid** [comment, marked as answer]
👋🏻 hey @gjtorikian hope you're well!

> Is there a pattern for accepting a list of `depends_on` for various versions of the same formula?

Nope, sorry. You need to make your own `postgis` formula if you want to do this. Our binaries will not work with different `postgres` versions.

**Reply by gjtorikian** [reply]
Darn, ok. Is it acceptable to just add a `postgis@15` formula, or is this something I should take up with the `postgis` developers? 

I hope you're well too!

**Reply by MikeMcQuaid** [reply]
@gjtorikian You can create one in your own tap but we won't consider it here, sorry! You could submit a PR to move `postgis` to use `postgresql@15` but I'm sufficiently out of the loop I can't make any promises that it'll be accepted.

**Reply by gjtorikian** [reply]
No worries! Thanks for the suggestion.
